### PR TITLE
[AE-535] Add unit tests for get_amp_tiles and get_spocs_and_direct_sold_tiles functions

### PR DESCRIPTION
**Pull request by akhan-mozilla**
## References

JIRA: [AE-535](https://mozilla-hub.atlassian.net/browse/AE-535)

## Description

This PR adds unit tests for the `get_amp_tiles` and `get_spocs_and_direct_sold_tiles` functions. The goal is to ensure that we correctly extract the expected keys and values from their responses and use them as intended in our views.

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [x] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/consvc-shepherd/blob/main/CONTRIBUTING.md).
- [x] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable).
- [ ] [Documentation](https://github.com/mozilla-services/consvc-shepherd/tree/main/docs) has been updated (if applicable).
- [x] Functional and performance test coverage has been expanded and maintained (if applicable).

[AE-535]: https://mozilla-hub.atlassian.net/browse/AE-535?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -83,7 +83,3 @@ add-ignore = ["D105","D107","D203", "D205", "D400"]
 # '^(?!migrations).*' ignores all Django-generated migration files
 match-dir = '^(?!migrations).*'
 
-[tool.coverage.run]
-omit = [
-    "consvc_shepherd/preview.py"
```
Comment (mashalifshin) [inline pyproject.toml:88]: Awesome! 🚀

**Review by mashalifshin — APPROVED** [review]
These look great, thank you for adding these! 

Just FYI, on a previous PR review of mine, Can mentioned this request mocking lib, https://github.com/jamielennox/requests-mock, which I have not tried out yet (I'm new to Python) but might make mocking nicer. So just wanted to let you know it's out there and feel free to use it if you think it's good

File: consvc_shepherd/tests/test_preview.py [hunk 2]
```diff
@@ -45,6 +56,194 @@
...
+            mockEnv, "US", "CA", True, mockEnv.spoc_site_id_mobile
+        )
+
+    def _test_get_spocs_and_direct_sold_tiles_case(
+        self,
+        mockEnv: Environment,
+        country: str,
+        region: str,
+        is_mobile: bool,
+        expected_site_id: int,
```
Comment (nanagb) [inline consvc_shepherd/tests/test_preview.py:183]: Small tweak: The type being passed into expected_site_id is "int | None"

**Comment by akhan-mozilla** [comment]
> These look great, thank you for adding these!
> 
> Just FYI, on a previous PR review of mine, Can mentioned this request mocking lib, https://github.com/jamielennox/requests-mock, which I have not tried out yet (I'm new to Python) but might make mocking nicer. So just wanted to let you know it's out there and feel free to use it if you think it's good

Thanks for sharing. With our current Implementation, using sideeffect functionality with the existing mocking setup should be sufficient for testing HTTP requests. However, this library appears to offer valuable features that we might consider incorporating in the future.

**Pull request merged**